swagger/smart_transfer_protocol_ws.yaml

```diff
@@ -0,0 +1,421 @@
+asyncapi: 2.5.0
+info:
+  title: ODF Smart Transfer Protocol
+  version: '0.1.0'
+  description: Open Data Fabric (ODF) extended protocol for dataset exchange between remote parties.
+  license:
+    name: Apache 2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0'
+
+channels:
+
+  dataset/pull/request:
+    subscribe:
+      operationId: odf.dataset.pull.request
+      summary: Initiates dataset pull from the ODF server
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPullRequest
+        name: DatasetPullRequest
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - knownHead          
+          properties:
+            datasetId:
+              type: string
+              format: did
+              description: Dataset identifier
+            knownHead:
+              type: string
+              format: multihash
+              description: Hash of the metadata head block known to the consumer
```

Comment:
Instead of `knownHead` perhaps we should make pull operate with a block interval `(tail, head]` (similarly to [MetadataBlock::iter_blocks_interval](https://github.com/kamu-data/kamu-cli/blob/master/kamu-core/src/domain/repos/metadata_chain.rs#L31)):
- `tail` (optional, exclusive) - tells the last block client already has (e.g. local `head` ref)
- `head` (optional, inclusive) - tells up to which point in the chain to sync (defaults to remote `head` ref, but in future can be a branch or a tag)

Reply:
Even without branches, setting a custom `head` is similar to checking out a particular commit in `git`:

```
  git checkout c123aca935
```

This sets the upper limit of synchronization, however it assumes the consumer already knows hashes of the new blocks.

In case of branches & tags, this would rather extend list of stored/exchanged references:
 - for branches store pair of inclusive references `[branchTail]` (first block of the branch) and `[branchHead]` (last block)
 - for tags store only the tagged block reference
 - allow symbolic references to branch/tag names in commands
 - synchronize lists of branch/tag references during pull/push
 - branch name likely needs to be passed on push/pull as well to choose which of the heads is impacted

Reply:
I've called the fields `beginAfter` and `stopAt` correspondingly to avoid confusing whose tail/head it is, consumer's or server's